examples/asana-googlecalendar/index.js

```diff
@@ -0,0 +1,183 @@
+const express = require("express");
+const asana = require("asana");
+const { google } = require("googleapis");
+const everyauth = require("@fusebit/everyauth-express");
+const moment = require("moment");
+const { v4: uuidv4 } = require("uuid");
+const cookieSession = require("cookie-session");
+const mustacheExpress = require("mustache-express");
+
+const app = express();
+const port = 3000;
+
+app.engine("mustache", mustacheExpress());
+app.set("view engine", "mustache");
+app.set("views", __dirname + "/views");
+app.use(express.json());
+app.use(express.urlencoded());
+app.use(cookieSession({ name: "session", secret: "secret" }));
+
+app.get("/", (req, res) => {
+  if (req.session.asanaUserId && req.session.googleUserId) {
+    return res.redirect(`/tasks`);
+  }
+  return res.render("index", { userId: uuidv4() });
+});
+
+// Asana Sign In Flow
+app.use(
+  "/asana/authorize/:userId",
+  everyauth.authorize("asana", {
+    finishedUrl: `/tasks`,
+    mapToUserId: (req) => req.params.userId,
+  })
+);
+
+// Google Sign In Flow
+app.use(
+  "/google/authorize/:userId",
+  everyauth.authorize("google", {
+    finishedUrl: `/tasks`,
+    mapToUserId: (req) => req.params.userId,
+  })
+);
+
+// Get userId from the authorization redirect or via session if already authorized.
+const handleSession = (req, res, next) => {
+  // Set Asana userId in session
+  if (req.query.serviceId === "asana") {
+    req.session.asanaUserId = req.query.userId;
+    return res.redirect("/google/authorize/" + req.session.asanaUserId);
+  }
+
+  // Set Google userId in session
+  if (req.query.serviceId === "google") {
+    req.session.googleUserId = req.query.userId;
+    return res.redirect("/tasks");
+  }
+
+  // If no userId in session, redirect to sign in page
+  if (!req.session.asanaUserId && !req.session.googleUserId) {
+    return res.redirect("/");
+  }
+
+  // If no Google userId in session, redirect to Google sign in page
+  if (!req.session.googleUserId && req.session.asanaUserId) {
+    return res.redirect("/google/authorize/" + req.session.asanaUserId);
+  }
+
+  // If no Asana userId in session, redirect to Google sign in page
+  if (!req.session.asanaUserId && req.session.googleUserId) {
+    return res.redirect("/asana/authorize/" + req.session.googleUserId);
+  }
+
+  console.log("Asana User ID:" + req.session.asanaUserId);
+  console.log("Google User ID:" + req.session.googleUserId);
+  return next();
+};
```

Comment:
Pull these out into the specific handlers for each endpoint so it's not all mixed in the same function.

Look for the values in the session, and if it's not there, then send the user through the signup phase from the top, just as a simple approach.